### ISSUE TITLE
Inconsistent date math support for reindexing api (source.index vs. dest.index)

### ISSUE THREAD

**Issue by ppf2** [elastic/elasticsearch#23533]
5.2

When running the reindex from remote api, if the source index name(s) are specified without encoding, it will complain about the /.  So for the source, it expects date math support index names to be URI encoded.

```
    "index": [
      "%3Cfilebeat-%7Bnow%2Fd-1d%7D%3E",
      "%3Cfilebeat-%7Bnow%2Fd%7D%3E"
    ]
```

But in the dest section of the reindex from remote api, when specifying URI encoded index names with date math support, it will throw the exception (because it expects the non-URI encoded version of the name):

```
        "type": "invalid_index_name_exception",
        "reason": "Invalid index name [%3Cfilebeat-consolidated-%7Bnow%7D%3E], must be lowercase",
```

It will be nice for source.index and dest.index to be consistent when date math is used.   If this is by design, it will be good to document this.

**Comment by ammar27** [comment]
Hi, I am new to contributing to open source projects and would like to try and fix this issue.

**Comment by bleskes** [comment]
@dnhatn are you still planning to work on this? I find it strange that the index names need to be encoded in the JSON.

**Comment by henningandersen** [comment]
Can confirm this issue still exists on master. Works fine for local indexes, but must escape for remote.

**Comment by henningandersen** [comment]
Fixed in 575918e8e6921bcfe8ae67020ff05f0ad4705605 such that source names used in reindex from remote can have date math characters too. Final fix in 8dbdd0688638ac2759da4e5026a3ceb0ac196048 to escape everything.
